manifests/config.pp

```diff
@@ -44,9 +44,21 @@
   }
 
   if $::foreman::use_foreman_service {
-    systemd::dropin_file { 'installer.conf':
-      unit    => "${::foreman::foreman_service}.service",
-      content => template('foreman/foreman.service-overrides.erb'),
+    $listen_socket = $foreman::foreman_service_bind ? {
+      Stdlib::IP::Address::V6 => "[${foreman::foreman_service_bind}]:${foreman::foreman_service_port}",
+      default                 => "${foreman::foreman_service_bind}:${foreman::foreman_service_port}",
+    }
+
+    systemd::dropin_file { 'foreman-socket':
+      filename => 'installer.conf',
+      unit     => "${::foreman::foreman_service}.socket",
+      content  => "[Service]\nListenSocket=\nListenSocket=${listen_socket}\n",
+    }
+
+    systemd::dropin_file { 'foreman-service':
+      filename => 'installer.conf',
```

Comment:
Was curious why we call this installer.conf ?

Reply:
We already did this before, but since you can't define `systemd::dropin_file` twice with 'installer.conf' I had to be explicit. This creates `/etc/systemd/system/foreman.service.d/installer.conf` and I thought this was the most logical name for it. `systemctl cat foreman` shows you how it would look like:

```text
# /usr/lib/systemd/system/foreman.service
[Unit]
Description=Foreman
Documentation=https://theforeman.org
After=network.target remote-fs.target nss-lookup.target
Requires=foreman.socket

[Service]
Type=simple
User=foreman
TimeoutSec=300
WorkingDirectory=/usr/share/foreman
ExecStart=/usr/share/foreman/bin/rails server --environment $FOREMAN_ENV --port $FOREMAN_PORT --binding $FOREMAN_BIND
Environment=FOREMAN_ENV=production FOREMAN_PORT=3000 FOREMAN_BIND=0.0.0.0

SyslogIdentifier=foreman

[Install]
WantedBy=multi-user.target

# /etc/systemd/system/foreman.service.d/installer.conf
[Service]
User=foreman
Environment=FOREMAN_ENV=production
Environment=FOREMAN_HOME=/usr/share/foreman
Environment=FOREMAN_BIND=127.0.0.1
Environment=FOREMAN_PORT=3000
Environment=FOREMAN_PUMA_THREADS_MIN=0
Environment=FOREMAN_PUMA_THREADS_MAX=16
Environment=FOREMAN_PUMA_WORKERS=2
```